aws/resource_aws_ec2_transit_gateway_vpc_attachment.go

```diff
@@ -246,15 +293,17 @@ func resourceAwsEc2TransitGatewayVpcAttachmentUpdate(d *schema.ResourceData, met
 			return fmt.Errorf("error describing EC2 Transit Gateway (%s): missing options", transitGatewayID)
 		}
 
-		if d.HasChange("transit_gateway_default_route_table_association") {
-			if err := ec2TransitGatewayRouteTableAssociationUpdate(conn, aws.StringValue(transitGateway.Options.AssociationDefaultRouteTableId), d.Id(), d.Get("transit_gateway_default_route_table_association").(bool)); err != nil {
-				return fmt.Errorf("error updating EC2 Transit Gateway Attachment (%s) Route Table (%s) association: %s", d.Id(), aws.StringValue(transitGateway.Options.AssociationDefaultRouteTableId), err)
+		if *transitGateway.OwnerId == vpcOwnerID {
```

Comment:
Given that it would be invalid to attempt adjusting these configuration arguments since we cannot manage the EC2 Transit Gateway Default Route Table for a RAM shared Transit Gateway, we can either remove all these proposed `resourceAwsEc2TransitGatewayVpcAttachmentUpdate` changes and allow the API to return any errors (I personally think this is preferable in this situation, paired with updates to the resource documentation) or continue with these adjustments where Terraform will instead just return a perpetual plan difference, e.g. `"true" => "false"`.

If we do keep these changes, we can remove the above `DescribeVpcs` API call above by using information present in the Terraform state here. 👍 We should also provide an inline comment for future code maintainers about this conditional.

```suggestion
		// We cannot modify Transit Gateway Route Tables for Resource Access Manager shared Transit Gateways
		if aws.StringValue(transitGateway.OwnerId) == d.Get("vpc_owner_id").(string) {
```